### PR TITLE
Fix for golint issues

### LINKED ISSUE #338: Fix the golint issues
/kind bug
/area provider/ibmcloud

**What steps did you take and what happened:**
Currently we are 42% per golint report per - https://goreportcard.com/report/github.com/kubernetes-sigs/cluster-api-provider-ibmcloud#golint, this issue is to fix or improve that number

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by Karthik-K-N**
**What this PR does / why we need it**:
This PR contain the changes to improve linting

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #338

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: cloud/scope/cluster.go [hunk 1]
```diff
@@ -278,15 +279,16 @@ func (s *ClusterScope) ensureSubnetUnique(subnetName string) (*vpcv1.Subnet, err
...
 func (s *ClusterScope) DeleteSubnet() error {
 	subnetID := *s.IBMVPCCluster.Status.Subnet.ID
 
 	// get the pgw id for given subnet, so we can delete it later
 	getPGWOptions := &vpcv1.GetSubnetPublicGatewayOptions{}
 	getPGWOptions.SetID(subnetID)
 	pgw, _, err := s.IBMVPCClients.VPCService.GetSubnetPublicGateway(getPGWOptions)
-	if pgw != nil && err == nil { // publicgateway found
-		// Unset the publicgateway for subnet first
+	if pgw != nil && err == nil { // publicg ateway found
```
Comment (mkumatag) [inline cloud/scope/cluster.go:290]:
```suggestion
	if pgw != nil && err == nil { // public gateway found
```

File: controllers/ibmvpcmachine_controller.go [hunk 2]
```diff
@@ -117,7 +117,7 @@ func (r *IBMVPCMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	if err != nil {
 		return ctrl.Result{}, errors.Errorf("failed to create scope: %+v", err)
 	}
-	// Always close the scope when exiting this function so we can persist any GCPMachine changes.
+	// Always close the scope when exiting this function, so we can persist any GCPMachine changes.
```
Comment (mkumatag) [inline controllers/ibmvpcmachine_controller.go:120]:
```suggestion
	// Always close the scope when exiting this function, so we can persist any IBMVPCMachine changes.
```

**Review by mkumatag — CHANGES_REQUESTED** [review]
few minor changes are required.

**Comment by Karthik-K-N** [comment]
All the review comments has been addressed

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**